### PR TITLE
Remove Python 2 compatibility code from Celery

### PR DESCRIPTION
**Pull request by thedrow**
This PR strips Python 2 completely from the code base except the `bin/` folder which will be replaced by a new implementation.

The only thing I'm still contemplating about is porting `reraise()` to `raise NewException() from exc`. I'd love to hear your thoughts on that one. There is a test that is commented out because of that.

I tried my best to make this PR reviewable.

### CODE REVIEW

**Comment by thedrow** [comment]
The tests pass locally and fail under a very specific condition.
I'll create an issue about this.

**Comment by thedrow** [comment]
> can we use the Python3.6 compatible release — PyPy3.6 v7.3.1 or allow failure for pypy3.6?

We can allow failures for PyPy3.6 but that means it will be overlooked.

On focal PyPy is not supported yet.
We'll probably move to Azure Pipelines as I planned. There's a CEP I need to write about that.

File: celery/utils/serialization.py [hunk 1]
```diff
@@ -137,7 +135,8 @@ def __init__(self, exc_module, exc_cls_name, exc_args, text=None):
         self.exc_cls_name = exc_cls_name
         self.exc_args = safe_exc_args
         self.text = text
-        Exception.__init__(self, exc_module, exc_cls_name, safe_exc_args, text)
+        Exception.__init__(self, exc_module, exc_cls_name, safe_exc_args,
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/utils/serialization.py:138]: I think we can call super.
That can be done on master.

[the remark this answers is not in the document]
Comment (auvipy) [reply celery/utils/serialization.py:138]: yes

**Comment by auvipy** [comment]
I just want to see the integration test running, so we can overlook pypy3 for few days until it's available again. I also started checking pypy internals on my local machine and checked update for the python3.7 branch.

**Comment by thedrow** [comment]
> I just want to see the integration test running, so we can overlook pypy3 for few days until it's available again. I also started checking pypy internals on my local machine and checked update for the python3.7 branch.

I'll work on that.

**Comment by auvipy** [comment]
[7733] /home/travis/build/celery/celery$ /home/travis/build/celery/celery/.tox/flake8/bin/flake8 -j 2 celery t

celery/backends/redis.py:126:5: F811 redefinition of unused '_reconnect_pubsub' from line 101

celery/backends/redis.py:141:5: F811 redefinition of unused 'reconnect_on_error' from line 116

celery/backends/database/__init__.py:73:47: E127 continuation line over-indented for visual indent

celery/utils/text.py:192:1: E305 expected 2 blank lines after class or function definition, found 1

celery/utils/time.py:247:43: E127 continuation line over-indented for visual indent

celery/bin/base.py:15:1: F401 'celery.five.PY2' imported but unused

celery/bin/base.py:15:1: F401 'celery.five.python_2_unicode_compatible' imported but unused

celery/app/defaults.py:73:60: E127 continuation line over-indented for visual indent

celery/worker/consumer/consumer.py:115:32: E127 continuation line over-indented for visual indent

t/unit/backends/test_dynamodb.py:432:5: F811 redefinition of unused 'test_prepare_put_request_with_ttl' from line 413

t/unit/utils/test_functional.py:155:5: E303 too many blank lines (2)

t/unit/utils/test_functional.py:168:5: E303 too many blank lines (2)

t/unit/utils/test_functional.py:182:5: E303 too many blank lines (2)

t/unit/utils/test_imports.py:49:1: E303 too many blank lines (3)

t/unit/utils/test_serialization.py:35:5: E303 too many blank lines (2)

**Comment by auvipy** [comment]
after the flake8 errors fixed we can merge it.

**Comment by thedrow** [comment]
flake8 errors are from mistakes during merge.
I'm fixing them right now.

File: celery/bin/base.py [hunk 2]
```diff
@@ -12,8 +12,7 @@
 
 from celery import VERSION_BANNER, Celery, maybe_patch_concurrency, signals
 from celery.exceptions import CDeprecationWarning, CPendingDeprecationWarning
-from celery.five import (PY2, getfullargspec, items, long_t,
-                         python_2_unicode_compatible, string, string_t,
+from celery.five import (getfullargspec, items, long_t, string, string_t,
```
Comment (auvipy) [inline celery/bin/base.py:15]: import from five still needed?

**Comment by thedrow** [comment]
> The only thing I'm still contemplating about is porting `reraise()` to `raise NewException() from exc`. I'd love to hear your thoughts on that one. There is a test that is commented out because of that.

@celery/core-developers Please weight in on this issue before we merge.

**Comment by auvipy** [comment]
allow me some more time to review once again tomorrow morning.

File: celery/bin/base.py (hunk 2 above)
Comment (thedrow) [reply celery/bin/base.py:15]: This is removed in the CLI refactor anyway.

Comment (auvipy) [reply celery/bin/base.py:15]: ok cool

**Comment by auvipy** [comment]
merging now. looking forward to reviewing the CLI refactor PR.

**Comment by thedrow** [comment]
> > The only thing I'm still contemplating about is porting `reraise()` to `raise NewException() from exc`. I'd love to hear your thoughts on that one. There is a test that is commented out because of that.
> 
> @celery/core-developers Please weight in on this issue before we merge.

@auvipy You didn't address this at all.

**Comment by auvipy** [comment]
The only thing I'm still contemplating about is porting reraise() to raise NewException() from exc. I'd love to hear your thoughts on that one. There is a test that is commented out because of that.

I tried my best to make this PR reviewable.

I missed that completely. I will check this out

**Pull request merged**